cloudformation/database/serverless-database.yaml

```diff
@@ -0,0 +1,344 @@
+    Type: String
+    NoEcho: true
+    # Only allow 'Medium' or better strength passwords according to https://dev.mysql.com/doc/refman/8.0/en/validate-password.html
+    AllowedPattern: (?=^.{8,}$)(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\w\s])^.*|^$
+    ConstraintDescription: Password must be BLANK or at least 8 characters with at least one uppercase letter, one lower case letter, one numeric digit, and one special (non-alphanumeric) character.
+    Default: ""
+  ClusterAdmin:
+    Description: Database cluster administrator user name.
+    Type: String
+    Default: clusteradmin
+    MinLength: 3
+    MaxLength: 64
+  Vpc:
+    Description: VPC ID to create the database cluster in (leave BLANK to create a new one).
+    Type: String
+    # Type: AWS::EC2::VPC::Id
+    Default: ""
+  DatabaseClusterSubnetOne:
+    Description: Subnet ID of the first subnet (leave BLANK to create).
+    Type: String
+    # Type: AWS::EC2::Subnet::Id
+    Default: ""
+  DatabaseClusterSubnetTwo:
+    Description: Subnet ID of the second subnet (leave BLANK to create). This subnet should be in different availability zone from the first subnet.
+    Type: String
+    # Type: AWS::EC2::Subnet::Id
+    Default: ""
+  MinCapacity:
+    Description: Minimum scaling capacity for the database cluster. Must be less than or equal to MaxCapacity.
+    Type: Number
+    Default: 1
+    AllowedValues:
+      - 1
+      - 2
+      - 4
+      - 8
+      - 16
+      - 32
+      - 64
+      - 128
+      - 256
+  MaxCapacity:
+    Description: Maximum scaling capacity for the database cluster. Must be greater than or equal to MinCapacity
+    Type: Number
+    Default: 4
+    AllowedValues:
+      - 1
+      - 2
+      - 4
+      - 8
+      - 16
+      - 32
+      - 64
+      - 128
+      - 256
+Transform: AWS::Serverless-2016-10-31
+Conditions:
+  CreateVPC: !Equals [!Ref Vpc, '']
+  CreateSubnets: !Or [!Condition CreateVPC, !Equals [!Ref DatabaseClusterSubnetOne, ''], !Equals [!Ref DatabaseClusterSubnetTwo, '']]
+  CreateAdminSecret: !Not [!Equals [!Ref AdminPasswordSecretString, '']]
+Rules:
+  EmptyVpcAssertions:
+    RuleCondition: !Equals
+      - !Ref Vpc
+      - ''
+    Assertions:
+      - Assert: !Equals
+          - !Ref DatabaseClusterSubnetOne
+          - ''
+        AssertDescription: DatabaseClusterSubnetOne must be empty if Vpc is not provided.
+      - Assert: !Equals
+          - !Ref DatabaseClusterSubnetTwo
+          - ''
+        AssertDescription: DatabaseClusterSubnetTwo must be empty if Vpc is not provided.
+Resources:
+  #
+  # Optional Networking
+  #
+  ServerlessDatabaseClusterVpc:
+    Type: 'AWS::EC2::VPC'
+    Condition: CreateVPC
+    Properties:
+      CidrBlock: 10.0.0.0/16
+      EnableDnsHostnames: true
+      EnableDnsSupport: true
+      InstanceTenancy: default
+      Tags:
+        - Key: Name
+          Value: ServerlessDatabaseClusterVPC
+        - Key: 'parallel-cluster:accounting'
+          Value: vpc
```

Comment:
Minor, here and elsewhere: I understand the reason why you're adding these tags and I agree. Nevertheless, I have the following observations:
1. Let's align key names with the ones used by ParallelCluster, i.e. `parallelcluster:something`, instead of `parallel-cluster:something`
2. We need a tag that declares a relation with the use case of the infrastructure. In this case the use case is Slurm Accounting, so I'd put a tag with name `parallelcluster:infrastructure` or `parallelcluster:usecase` and value  `Slurm Accounting`. We could replicate this in any other 1-click template , e.g. the one with the AD infrastructure would have `parallelcluster:usecase=Multi User`
3. I would avoid tags like `parallelcluster:accounting=resource` (e.g. `parallelclluster:accountig=vpc` becausde the type of resource is implicit in the resource itself and it does not need a tag for it.

Reply:
Yes, this sounds good.